src/NServiceBus.Core/Pipeline/Incoming/DeserializeLogicalMessagesConnector.cs

```diff
@@ -96,7 +95,9 @@ List<LogicalMessage> Extract(IncomingMessage physicalMessage)
             using (var stream = new MemoryStream(physicalMessage.Body))
             {
                 var messageTypes = messageMetadata.Select(metadata => metadata.MessageType).ToList();
-                var messageSerializer = deserializerResolver.Resolve(physicalMessage.Headers[Headers.ContentType]);
+                string contentType;
+                physicalMessage.Headers.TryGetValue(Headers.ContentType, out contentType);
+                var messageSerializer = deserializerResolver.Resolve(contentType);
```

Comment:
Also wonder if it makes sense to set the content type header if it was not present and we succeed at deserializing with the default ? 
This would make downstream callers that need to look it up happy